src/kernel/backed_heap.c

```diff
@@ -38,7 +34,7 @@ static void physically_backed_dealloc(heap h, u64 x, bytes length)
 
 static u64 physically_backed_alloc(heap h, bytes length)
```

Comment:
There's some redundancy here, and it seems like physically_backed_alloc and physically_backed_dealloc should be replaced with wrappers that call backed_alloc_map and backed_dealloc_unmap, respectively. If the backed_* functions are made inline and only conditionally set *phys if phys == 0, then the compiler should be able to factor out this conditional. The inline shouldn't affect the vft assignment, so these functions will be inlined within this file and also have incarnate forms that are referenced by the vft.

Reply:
OK, I made the the backed_* functions static inline, and they are now called by the existing alloc/dealloc functions, so there is no code duplication.